Add Dockerfile for the bi connector

jdavisyei opened the pull request:
Adding the BI connector dockerfile that Mathias sent over.

Commit:
Add Dockerfile for the bi connector

Commit:
Add a note about only using the docker-compose file on local

Commit:
Enable SSL to be turned on

Commit:
Add certificate logic

Commit:
Fix url to cert

Commit:
Try tailing the log

Commit:
Remove extra tail and swap the cert/key

Commit:
Try removing the pem key password arg to see if that's why it won't start

Commit:
Try bash job control

Commit:
Add some logging

Commit:
Try turning off SSL to see what it does

Commit:
Fix command line flag

Commit:
Cleanup the run script a little

Commit:
Add env var so we can inject extra options without rebuilding

Commit:
Set read preference to secondary

Commit:
Remove named pipe and just read the file directly

Commit:
Add replica set config

Commit:
Echo URI

Commit:
Tweak log

Commit:
Didn't work so removing it

Commit:
Remove read preference to see if it starts working again as it appears to be crashing

Commit:
Fix tail not finding file

AAkindele commented on mdb-bi-connector/Dockerfile:
```diff
@@ -0,0 +1,16 @@
+FROM debian:stretch
+MAINTAINER Matthias Granberry <matthias.granberry@youearnedit.com>
```
the docker docs say this is deprecated. also, can this be an alias email? don't want to jinx it, but just in case someone leaves.

jdavisyei replied:
Removed it since we really don't need it. It seems like they deprecated it because it was pointless.

AAkindele commented on mdb-bi-connector/docker-compose.yml:
```diff
@@ -0,0 +1,26 @@
+    command: run.sh
+    environment:
+      MONGO_AUTH: 1
+      MONGO_HOST : "${MONGODB_HOST}"
```
🤔 💭 what is this interpolation format doing? is it getting the value from the host environment variable?

jdavisyei replied:
That is for local, I'll just delete the file. We don't need it. It's to pull in an environment variable.

jdavisyei committed:
Removing the docker-compose file

We don't need the docker-compose file in this repo. I just inadvertently added it from Mathiases zip. It isn't really relevant to the docker file itself.

jdavisyei committed:
Remove matintainer since it's non essential and deprecated
